src/aws-cpp-sdk-core/include/aws/core/auth/GeneralHTTPCredentialsProvider.h

```diff
@@ -0,0 +1,105 @@
+             * @return true if provider configuration is valid
+             */
+            bool IsValid() const
+            {
+                return m_ecsCredentialsClient != nullptr;
+            }
+
+            /**
+             * Initializes the provider to retrieve credentials from the ECS metadata service every 5 minutes,
+             * or before it expires.
+             * @param resourcePath A path appended to the metadata service endpoint.
+             * @param refreshRateMs The number of milliseconds after which the credentials will be fetched again.
+             */
+            // TODO: 1.12: AWS_DEPRECATED("This c-tor is deprecated, please use one above.")
+            GeneralHTTPCredentialsProvider(const char* resourcePath, long refreshRateMs = REFRESH_THRESHOLD)
+              : GeneralHTTPCredentialsProvider(resourcePath, "", "", "", refreshRateMs)
+            {}
+
+            /**
+             * Initializes the provider to retrieve credentials from a provided endpoint every 5 minutes or before it
+             * expires.
+             * @param endpoint The full URI to resolve to get credentials.
+             * @param token An optional authorization token passed to the URI via the 'Authorization' HTTP header.
+             * @param refreshRateMs The number of milliseconds after which the credentials will be fetched again.
+             */
+            // TODO: 1.12: AWS_DEPRECATED("This c-tor is deprecated, please use one above.")
+            GeneralHTTPCredentialsProvider(const char* endpoint, const char* token, long refreshRateMs = REFRESH_THRESHOLD)
+              : GeneralHTTPCredentialsProvider("", endpoint, token, "", refreshRateMs)
+            {}
+
+            /**
+             * Initializes the provider to retrieve credentials using the provided client.
+             * @param client The ECSCredentialsClient instance to use when retrieving credentials.
+             * @param refreshRateMs The number of milliseconds after which the credentials will be fetched again.
+             */
+            GeneralHTTPCredentialsProvider(const std::shared_ptr<Aws::Internal::ECSCredentialsClient>& client,
+                    long refreshRateMs = REFRESH_THRESHOLD);
+            /**
+            * Retrieves the credentials if found, otherwise returns empty credential set.
+            */
+            AWSCredentials GetAWSCredentials() override;
+
+            static const char AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE[];
+            static const char AWS_CONTAINER_CREDENTIALS_RELATIVE_URI[];
+            static const char AWS_CONTAINER_CREDENTIALS_FULL_URI[];
+            static const char AWS_CONTAINER_AUTHORIZATION_TOKEN[];
+
+        protected:
+            void Reload() override;
+
+        private:
+            bool ExpiresSoon() const;
+            void RefreshIfExpired();
+
+            std::shared_ptr<Aws::Internal::ECSCredentialsClient> m_ecsCredentialsClient;
```

Comment:
nit: should we do something like

```
using GeneralHttpClient = ECSCredentialsClient;
```

like you down below now that after this is, this really doesnt have anything to do with ecs anymore?

Reply:
more so im asking if we should create an alias for this now because the name "ecsClient" doesnt reflect that its not actually calling ECS nessecarily.